test.e2e/test-server.js

```diff
@@ -81,17 +81,28 @@ masterApp.post('/setup', function(req, res, next) {
 
   lbApp = loopback();
 
-  lbApp.dataSource('db', { connector: 'memory', defaultForType: 'db' });
-  lbApp.dataSource('mail', { connector: 'mail', defaultForType: 'mail' });
+  lbApp.dataSource('db', { connector: 'memory' });
+  for (var m in initialModels) {
+    if (initialModels[m].base.modelName == 'PersistedModel')
+      lbApp.model(initialModels[m], { dataSource: 'db' });
+  }
 
   for (var m in models) {
-    models[m].dataSource = 'db';
-    var model = initialModels[m];
-    lbApp.model(model || m, models[m]);
+    var model = null;
+    var options = models[m].options || {};
+    if (initialModels[m]) {
+      model = initialModels[m];
+      lbApp.model(model, extend({ dataSource: 'db' }, options));
```

Comment:
If I am reading this code correctly, then you are calling `lbApp.model` twice for any custom model that happens to be in `initialModels` too.

I think a better solution is to merge `initialModels` to `models` so that you can iterate only over `models` here.